### PR TITLE
PYTHON-3005 Make maxConnecting configurable

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -184,7 +184,8 @@ Notable improvements
 ....................
 
 - Enhanced connection pooling to create connections more efficiently and
-  avoid connection storms.
+  avoid connection storms. Added the ``maxConnecting`` URI and
```
Comment (behackett) [inline doc/changelog.rst:187]: I know these two things are related, but I would suggest making them separate bullet points.

**Review by behackett — APPROVED** [review]
Looking at the existing code for limiting the number of concurrent connection attempts, it's almost like you expected this to become a configurable. :-)

File: doc/changelog.rst (hunk 1 above)
Comment (ShaneHarvey) [reply doc/changelog.rst]: Done.

**Review by ShaneHarvey — COMMENTED** [review]
Ready for another look.

IIRC when I implemented this we were still discussing whether to make it configurable or not.

**Comment by behackett** [comment]
Still looks good to me.

**Pull request merged**